app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers/target_collection.rb

```diff
@@ -65,6 +65,18 @@ def sshkeys
     []
   end
 
+  def snapshots
+    @snapshots ||= references(:vms).map do |ems_ref|
+      snapshots_api.pcloud_cloudinstances_snapshots_getall(cloud_instance_id).snapshots.select do |snapshot|
+        snapshot.pvm_instance_id == ems_ref
+      end
+    rescue IbmCloudPower::ApiError => err
+      error_message = JSON.parse(err.response_body)["description"]
+      _log.debug("SnapshotID not found: #{error_message}")
```

Comment:
Would the issue here be that the snapshot_id isn't found?  Only the `cloud_instance_id` is actually passed in as a parameter.

Reply:
Good catch. I updated the message to indicate it could be any error from the get snapshots api call.